mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -1,1376 +0,0 @@
-//===- rocm-runtime-wrappers.cpp - MLIR ROCM runner wrapper library -------===//
```

Comment:
... hold on, don't we still use these functions in tests? @sjw36 

I'm surprised this removal didn't break the build

Reply:
Now the tests are using mlir_rocm_runtime from upstream according to a recent commit PR631

Reply:
Not all the functions the tests need are in that file though

Reply:
Thank you for pointing that out. I'll try to fix it.

Reply:
The question will be ... do the tests actually run with this file deleted?

Reply:
No. It turns out I was testing without removing the rocm-runtime-wrapper library.
I brought it back and will submit again soon after the testing is done.

Reply:
Potentially I'd like to see us not having to maintain an additional "rocm-runtime-wrapper` (versus the upstream duplicate). I think what can be done (after this PR is merged) is:
 - file an upstream PR to add in whatever is necessary there
 - file another PR to get rid of `rocm-runtime-wrapper` duplicate in our side

Reply:
@krzysz00 Good point. I just forgot about the fact that validation related code are in here as well. 

So let me rephrase, in the upcoming refactor to decouple these two:
 - all `mgpu.*` function moved to external repo
 - all `mcpu.*` function and its utils remain (and potentially, we can put this to a new runtime library `rocm-validation-wrapper`)

Just so that we are clear on the boundary of the two components.